### PR TITLE
Tre rail 844 missing sorting in bucket interface

### PR DESCRIPTION
**Pull request by tomasreichmann**

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/132/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1379/ : SUCCESS in 2m 59s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/568/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/133/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/485/ : UNSTABLE in 14m 48s

**Comment by tomasreichmann** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/135/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1381/ : SUCCESS in 1m 32s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/569/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/136/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/486/ : UNSTABLE in 14m 47s

**Comment by tomasreichmann** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/140/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1384/ : SUCCESS in 4m 02s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/571/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/141/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/488/ : SUCCESS in 14m 53s

File: src/components/afm/FunnelChart.tsx [hunk 1]
```diff
@@ -12,7 +12,7 @@ export {
 import { ICommonChartProps } from '../core/base/BaseChart';
 import { FunnelChart as CoreFunnelChart } from '../core/FunnelChart';
 
-function generateDefaultDimensions(afm: AFM.IAfm): AFM.IDimension[] {
+export function generateFunnelDimensions(afm: AFM.IAfm): AFM.IDimension[] {
```
Comment (lhorak) [inline src/components/afm/FunnelChart.tsx]: It seems like you've moved all the dimensions generators to helpers, why did you leave this one in the file?

File: src/helpers/tests/resultSpec.spec.ts [hunk 2]
```diff
@@ -0,0 +1,198 @@
+// (C) 2007-2018 GoodData Corporation
+import { VisualizationObject, AFM } from '@gooddata/typings';
+
+// import { VisualizationTypes } from '../../constants/visualizationTypes';
```
Comment (lhorak) [inline src/helpers/tests/resultSpec.spec.ts]: this should be deleted if not used

File: src/helpers/tests/resultSpec.spec.ts [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+                        measureIdentifier: 'm1'
+                    }
+                }]
+            }
+        };
+
+        const expectedResultSpec: AFM.IResultSpec = {
+            dimensions: [
+                {
+                    itemIdentifiers: [
```
Comment (lhorak) [inline src/helpers/tests/resultSpec.spec.ts]: could be one liner, as in other cases :)

File: stories/core_components/BarChart.tsx [hunk 4]
```diff
@@ -5,12 +5,22 @@ import { screenshotWrap } from '@gooddata/test-storybook';
...
+    ATTRIBUTE_1,
+    ATTRIBUTE_2,
+    MEASURE_1,
+    MEASURE_1_WITH_ALIAS,
+    MEASURE_2,
+    ATTRIBUTE_1_SORT_ITEM,
+    MEASURE_2_SORT_ITEM
+} from '../data/componentProps';
+
+const dimensions = { width: 800, height: 400 };
```
Comment (lhorak) [inline stories/core_components/BarChart.tsx]: How about renaming this object to eg wrapperStyle and then passing it to each style prop istead of destructuring it?

**Comment by tomasreichmann** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/148/ : SUCCESS in 17s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1393/ : FAILURE in 2m 06s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/576/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/149/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/490/ : UNSTABLE in 19m 08s

**Comment by tomasreichmann** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/154/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1397/ : SUCCESS in 3m 48s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/578/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/155/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/492/ : UNSTABLE in 17m 34s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/158/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1401/ : FAILURE in 7m 07s

**Comment by tomasreichmann** [comment]
ok to test

**Comment by yenkins** [comment]
Link to Storybook test result:

https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/580/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/159/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-react-components-pull-request-dispatcher/493/ : SUCCESS in 22m 51s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/160/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1403/ : SUCCESS in 2m 08s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/24/ : SUCCESS in 14s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/1404/ : SUCCESS in 1m 47s
- https://ci.intgdc.com/job/gooddata-react-components-storybook-zuul-docker/582/ : SUCCESS in 15m 30s

**Pull request merged**